File: 0463-island-perimeter/0463-island-perimeter.java
class Solution {
    public int periCalculate(int[][] grid, int row, int col){
        int n = grid.length;
        int m = grid[0].length;
        int perimeter = 0;
        if(row == 0 || grid[row - 1][col] != 1)perimeter++;
        if(col == 0 || grid[row][col - 1] != 1)perimeter++;
        

        if(row == n - 1 || grid[row + 1][col] != 1)perimeter++;
        if(col == m - 1 || grid[row][col + 1] != 1)perimeter++;

        return perimeter;
    }
    public int islandPerimeter(int[][] grid) {
        int n = grid.length;
        int m = grid[0].length;

        int perimeter = 0;

        for(int i = 0; i < n; i++){
            for(int j = 0; j < m; j++){
                if(grid[i][j] == 1){
                    perimeter += periCalculate(grid, i, j);
                }
            }
        }
        return perimeter;
    }
}

File: 0463-island-perimeter/0463-island-perimeter.js
/**
 * @param {number[][]} grid
 * @return {number}
 */

 const pericalculate = function(grid, row, col){
    const n = grid.length;
    const m = grid[0].length;
    let perimeter = 0;
    if(row == 0 || grid[row - 1][col] != 1)perimeter++;
    if(col == 0 || grid[row][col - 1] != 1)perimeter++;
    
    if(row == n - 1 || grid[row + 1][col] != 1)perimeter++;
    if(col == m - 1 || grid[row][col + 1] != 1)perimeter++;
    return perimeter;
 }
var islandPerimeter = function(grid) {
    const n = grid.length;
    const m = grid[0].length;
    let perimeter = 0;

    for(let i = 0; i < n; i++){
        for(let j = 0; j < m; j++){
            if(grid[i][j] == 1){
                perimeter += pericalculate(grid, i, j);
            }
        }
    }
    return perimeter;
};